app/src/androidTest/java/ch/sdp/vibester/BuzzerScoreUpdaterTest.kt

```diff
@@ -6,25 +6,43 @@ import org.junit.Test
 
 public class BuzzerScoreUpdaterTest {
 
-    // FIXME: Uncomment this when the infinite loop issue in the class is fixed
-/*
     @Test
     fun constructorTestWithSameSizedArrays() {
-        val idArray = arrayOf(0, 1, 2, 3)
+        val idArray = arrayOf(R.id.buzzer_0, R.id.buzzer_1, R.id.buzzer_2, R.id.buzzer_3)
         val scoreArray = arrayOf(0, 0, 0, 0)
         val expectedMap = LinkedHashMap<Int, Int>()
-        expectedMap.put(0, 0)
-        expectedMap.put(1, 0)
-        expectedMap.put(2, 0)
-        expectedMap.put(3, 0)
-        val testUpdater = BuzzerScoreUpdater(scoreArray, idArray)
-        assert(testUpdater.getMap()==(expectedMap)) // value or reference?
+        expectedMap.put(R.id.buzzer_0, 0)
+        expectedMap.put(R.id.buzzer_1, 0)
+        expectedMap.put(R.id.buzzer_2, 0)
+        expectedMap.put(R.id.buzzer_3, 0)
+        val testUpdater = BuzzerScoreUpdater(idArray, scoreArray)
+        for (key in idArray) {
+            assert(testUpdater.getMap().keys.contains(key))
+        }
+        for (value in scoreArray) {
+            assert(testUpdater.getMap().values.contains(value))
+        }

```

Comment:
I was searching and I believe  it is not correct to use java's assert in the test. Because in this case the java's exception will appear, not junit's exception. I would change assert on junit's assertTrue or assertEquals.

Reply:
Also here ExpectedMap is not used, so it can be deleted

Reply:
Good point, I'll change this!